sky/clouds/azure.py

```diff
@@ -294,7 +294,8 @@ def make_deploy_resources_variables(
         else:
             custom_resources = None
 
-        if resources.image_id is None:
+        if resources.image_id is None or resources.extract_docker_image(
+        ) is not None:
```

Comment:
```suggestion
        if (resources.image_id is None or resources.extract_docker_image(
        ) is not None):
```
Wrapping like this + formatting would look better :)